c10/util/BFloat16-inl.h

```diff
@@ -0,0 +1,19 @@
+#pragma once
+
+#include <c10/macros/Macros.h>
+
+namespace c10 {
+
+/// Constructors
+
+inline C10_HOST_DEVICE BFloat16::BFloat16(float value) {
+  val_ = detail::bits_from_f32(value);
```

Comment:
These are marked `C10_HOST_DEVICE` but call host-only functions. That's not going to work when you include this in a CUDA file.

Reply:
Suggest to use round-nearest-to-even (RNE) as the default rounding mode rather than truncating. From our experiments, RNE is essential to guarantee convergence. Moreover, Tensorflow also uses RNE by default:

https://github.com/tensorflow/tensorflow/blob/a650bdc7e68285179da8abc310d5ffda268230a9/tensorflow/core/lib/bfloat16/bfloat16.h#L66
https://github.com/tensorflow/tensorflow/blob/a650bdc7e68285179da8abc310d5ffda268230a9/tensorflow/core/lib/bfloat16/bfloat16.h#L184

Reply:
@jgong5 we should just make this consistent with our other rounds.  I believe we are doing half-to-even as of https://github.com/pytorch/pytorch/pull/17443.

I think this can be implemented in another PR though.